server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -166,6 +169,18 @@ public final class IndexModule {
         Property.NodeScope
     );
 
+    /** Which lucene file extensions to load with nio. All others will default to mmap. Takes precedence over {@link #INDEX_STORE_HYBRID_MMAP_EXTENSIONS}.
+     *  This is an expert setting.
+     *  @see <a href="https://lucene.apache.org/core/9_5_0/core/org/apache/lucene/codecs/lucene95/package-summary.html#file-names">Lucene File Extensions</a>.
+     */
+    public static final Setting<List<String>> INDEX_STORE_HYBRID_NIO_EXTENSIONS = Setting.listSetting(
```

Comment:
@dzane17 it looks pretty good, I think the only one thing we need to add is validation when both INDEX_STORE_HYBRID_NIO_EXTENSIONS and INDEX_STORE_HYBRID_MMAP_EXTENSIONS settings are used (since in this case we essentially ignore the `INDEX_STORE_HYBRID_MMAP_EXTENSIONS`), could you please add that and raise an exception if such a case is detected?

Reply:
I added an exception and updated the unit test